toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -193,11 +197,46 @@ func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, ou
 	)
 
 	logBaseName := filepath.Base(srpmFile) + ".log"
+	const checkAttempts = 3
+	totalAttempts := buildAttempts
+	if agent.Config().RunCheck && totalAttempts < checkAttempts {
+		totalAttempts = checkAttempts
+	}
 	err = retry.Run(func() (buildErr error) {
 		builtFiles, logFile, buildErr = agent.BuildPackage(srpmFile, logBaseName, outArch, dependencies)
-		return
-	}, buildAttempts, retryDuration)
+		// If the package builds with no errors and RUN_CHECK=y, check logs to see if the %check section passed, and if not, return as the build error.
+		if buildErr != nil {
+			return
+		}
 
+		if agent.Config().RunCheck {
+			file, logErr := os.Open(logFile)
+			// If we can't open the log file, that's a build error.
+			if logErr != nil {
+				logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logfile, logErr)
+				buildErr = logErr				
+				return
+			}
+			defer file.Close()
+
+			for scanner := bufio.NewScanner(file); scanner.Scan() {
+				currLine := scanner.Text()
+				// Anything besides 0 is a failed test
+				if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
+					failedLogFile := fmt.Sprintf("%s-FAILED_TEST-%d.log", filepath.Base(srpmFile), time. Now().UnixMilli())
```

Comment:
Looks like I made an autocorrect error. Sorry about that.
```suggestion
					failedLogFile := fmt.Sprintf("%s-FAILED_TEST-%d.log", filepath.Base(srpmFile), time.Now().UnixMilli())
```